lib/graphlient/adapters/faraday_adapter.rb

```diff
@@ -0,0 +1,24 @@
+require "faraday"
+
+module Graphlient
+  module Adapters
+    class FaradayAdapter
+      def initialize(url, headers: {})
+        @conn = Faraday.new(url: url, headers: headers)  do |c|
```

Comment:
You should be using `attr_accessor` or a private connection method that initializes the connection on first call, and avoid local `@variables` altogether.